pkg/ipamd/datastore/data_store.go

```diff
@@ -841,6 +1112,39 @@ func (ds *DataStore) FreeableIPs(eniID string) []string {
 	return freeable
 }
 
+// FreeablePrefixes returns a list of unused and potentially freeable IPs.
+// Note result may already be stale by the time you look at it.
+func (ds *DataStore) FreeablePrefixes(eniID string) []string {
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+
+	eni := ds.eniPool[eniID]
+	if eni == nil {
+		// Can't free any IPs from an ENI we don't know about...
+		return []string{}
+	}
+	ds.log.Infof("In freeable prefix for eni %s", eniID)
+
+	freeable := make([]string, 0, len(eni.IPv4Prefixes))
+	for _, prefix := range eni.IPv4Prefixes {
+		DBlen := (int)(prefix.AllocatedIPs.IPsPerPrefix / 8)
+
+		ds.log.Infof("In get IP from prefix - %s and octet len %d", prefix, DBlen)
```

Comment:
This is not `Info` worthy - particularly inside a loop like this.  Demote to `Debug` or remove.